crates/compiler/fmt/src/expr.rs

```diff
@@ -1715,7 +1697,11 @@ fn fmt_when<'a>(
 
         buf.indent(indent + INDENT);
         let line_indent = buf.cur_line_indent();
-        buf.push_str(" ->");
+        if buf.ends_with_space() || buf.ends_with_newline() {
```

Comment:
Pro tip: you should be able to just use `buf.ensure_ends_with_whitespace()`, followed by `buf.push_str("->")`